add mstarpy to new terminal version

Mael-J opened the pull request:
# Description

- [ ] Summary of the change / bug fix.
- [ ] Link # issue, if applicable.
- [ ] Screenshot of the feature or the bug before/after fix, if applicable.
- [ ] Relevant motivation and context.
- [ ] List any dependencies that are required for this change.


# How has this been tested?

* Please describe the tests that you ran to verify your changes.
* Provide instructions so we can reproduce.
* Please also list any relevant details for your test configuration.
- [ ] Make sure affected commands still run in terminal
- [ ] Ensure the SDK still works
- [ ] Check any related reports


# Checklist:

- [ ] Update [our Hugo documentation](https://openbb-finance.github.io/OpenBBTerminal/) following [these guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/website).
- [ ] Update our tests following [these guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/tests).
- [ ] Make sure you are following our [CONTRIBUTING guidelines](https://github.com/OpenBB-finance/OpenBBTerminal/blob/main/CONTRIBUTING.md).
- [ ] If a feature was added make sure to add it to the corresponding [scripts file](https://github.com/OpenBB-finance/OpenBBTerminal/tree/main/scripts).


# Others
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.

jmaslek commented:
Hey @Mael-J thanks for rebasing this an porting over to 2.0.

I'm a little confused.  When I run poetry install to add the package I get the following:
```
Package operations: 3 installs, 1 update, 2 removals

  • Removing setuptools (65.4.1)
  • Removing wheel (0.38.4)
  • Installing textwrap3 (0.9.2)
  • Installing ansiwrap (0.8.4)
  • Installing mstarpy (0.0.1)
```

And when I go into the terminal to enter funds, I get 
```
2022 Dec 09, 14:18 (🦋) / $ funds

Error: No module named 'pkg_resources'
```

Which is a result of removing setuptools.  I can get around this by installing setuptools, but this is some strange behavior.  On the main branch, `poetry install` does not delete setuptools and wheel, so was anything else added to poetry on your end?

Will dive into starting a review now, just wanted to bring that up at first.

jmaslek commented:
I am going to have a few requests here that I will go through in the code, but I am going to report some bugs in comments like this.

```
2022 Dec 09, 14:38 (🦋) /funds/ $ load_ms --fund FOUSA00L8W

The funds Vanguard 500 Index Fund Admiral Shares - 52.8.VFIAX (FOUSA00L8W) from the country us is loaded
```
Get unhandled error with  the following:
```
2022 Dec 09, 14:39 (🦋) /funds/ $ holdings -t bond

Error: "None of [Index(['isin', 'securityName', 'weighting', 'country'], dtype='object')] are in the "

Error: 'list' object has no attribute 'columns'
```

jmaslek commented:
On the plot:  it would be desired to just default to showing price for the loaded fund:  

There could then be additional flags to show the following:
- performance
- index
- category
<img width="780" alt="Screenshot 2022-12-09 at 2 48 11 PM" src="https://user-images.githubusercontent.com/18151143/206785181-5b4c9336-984f-4af9-8ffb-7f405a70894d.png">


<img width="780" alt="Screenshot 2022-12-09 at 2 48 11 PM" src="https://user-images.githubusercontent.com/18151143/206785176-691d5fc8-f3e1-4f32-a847-ecf87035185c.png">

<img width="780" alt="Screenshot 2022-12-09 at 2 48 11 PM" src="https://user-images.githubusercontent.com/18151143/206785069-c2188baf-875d-4567-bb8a-c88c12d7e669.png">

jmaslek commented:
All of the `[Investing]`/investpy functions should be removed.  And then your functions should not have `_ms`.  

For example: there is a search and search_ms.  Get rid of the search and rename your search_ms -> search.  Repeat for each other instance of _ms.

jmaslek commented:
Also the load function should have a start/end date option.

jmaslek commented on openbb_terminal/mutual_funds/mutual_fund_controller.py:
```diff
@@ -552,3 +584,233 @@ def call_forecast(self, _):
+            prog="country",
+            description="Set a country for funds",
+        )
+        parser.add_argument(
+            "-n",
+            "--name",
+            type=str,
+            dest="name",
+            nargs="+",
+            help="country to select",
+            default="",
```
Note we have a common convention across the terminal for countries where we enter `united_states` instead of `united states`. The reason is for continuity so you dont have to remember which menus are one way or the other.

jmaslek reviewed (changes requested):
We also would like to see a good level of unit test coverage on the _model functions as well as a few examples on the _controller.  We also will need some integration tests which reside in the scripts folder.

Mael-J commented:
Hello,

in the last commit, I changed the country names, add date option to the load function, correct the holdings methods in case of no assets displayed, removed the Investing source.

I will check how to do the tests.

About poetry, I just added mstarpy with poetry add mstarpy.

Thank you

jmaslek commented on openbb_terminal/mutual_funds/mutual_fund_controller.py:
```diff
@@ -47,12 +45,15 @@ class FundController(BaseController):
-        "alswe",
-        "infoswe",
-        "forecast",
+        # "alswe",
+        # "infoswe",
```
These two can be left in.  But only showing when sweden is selected (like it was)

jmaslek commented on openbb_terminal/mutual_funds/mutual_fund_controller.py:
```diff
@@ -137,23 +106,423 @@ def print_help(self):
+            # return ["funds", f"load {self.fund_name} --name"]
         return []
 
+    # @log_start_end(log=logger)
+    # def call_country(self, other_args: List[str]):
+    #     """Process country command"""
+    #     parser = argparse.ArgumentParser(
+    #         add_help=False,
+    #         formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    #         prog="country",
+    #         description="Set a country for funds",
+    #     )
+    #     parser.add_argument(
+    #         "-n",
+    #         "--name",
+    #         type=str,
+    #         dest="name",
+    #         nargs="+",
+    #         help="country to select",
+    #     )
+    #     if other_args and "-" not in other_args[0][0]:
+    #         other_args.insert(0, "-n")
+    #     ns_parser = self.parse_known_args_and_warn(parser, other_args)
+    #     if ns_parser:
+    #         country_candidate = " ".join(ns_parser.name)
+    #         if country_candidate.lower() in self.fund_countries:
+    #             self.country = " ".join(ns_parser.name)
+    #         else:
+    #             console.print(
+    #                 f"{country_candidate.lower()} not a valid country to select."
```
We would prefer to just get rid of everything and not comment it out.  We can always get it back from the git history

jmaslek commented on openbb_terminal/mutual_funds/mutual_fund_controller.py:
```diff
@@ -137,23 +106,423 @@ def print_help(self):
+    #         )
+    #     return self.queue
+
+    @log_start_end(log=logger)
+    def call_overview(self, other_args: List[str]):
+        """Process country command"""
+        parser = argparse.ArgumentParser(
+            add_help=False,
+            formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+            prog="overview",
+            description="Show overview of funds from selected country.",
+        )
+        parser.add_argument(
+            "-l",
+            "--limit",
+            help="Number of search results to show",
+            type=check_positive,
+            dest="limit",
+            default=10,
+        )
+        ns_parser = self.parse_known_args_and_warn(
+            parser, other_args, export_allowed=EXPORT_ONLY_RAW_DATA_ALLOWED
+        )
+        if ns_parser:
+            investpy_view.display_overview(
+                self.country, limit=ns_parser.limit, export=ns_parser.export
+            )
+        return self.queue
```
This should also be taken out.  Or if it still works, it can be added to the choices.

jmaslek commented:
Thanks for making the changes so quickly.

I am getting error on running equity:

```

2022 Dec 14, 19:15 (🦋) /funds/ $ load FOUSA00L8W

The funds Vanguard 500 Index Fund Admiral Shares - 52.8.VFIAX (FOUSA00L8W) from the country United_States is loaded

2022 Dec 14, 19:15 (🦋) /funds/ $ equity

Error: 'equityHoldings'
```

jmaslek commented on openbb_terminal/mutual_funds/mutual_fund_controller.py:
```diff
@@ -225,330 +581,192 @@ def call_search(self, other_args: List[str]):
         )
 
+        parser.add_argument(
+            "-comp",
```
By convention, we just use one letter with a single dash, so `-c` should be used here

jmaslek commented:
And then can the `holdings` function have the same compare options as the plot function?

Mael-J commented:
I deleted the equity command which is not working anymore and replaced by the command holdings -t equity. 

What do you mean about the compare options for the holdings ? MorningStar does not display holdings of the index or the category. I can rethink about the mutual_funds menu where the user select a list of funds and can compare their performances, holdings and ESG metrics

jmaslek commented:
> What do you mean about the compare options for the holdings ? MorningStar does not display holdings of the index or the category. I can rethink about the mutual_funds menu where the user select a list of funds and can compare their performances, holdings and ESG metrics

Apologies for the delay over the holiday break!  By the compare - I believe I was referring to it showing the benchmark as well as the fund holding by default -- having a flag like on the plot to show this

hjoaquim commented:
Closing in favor of  #4068
